Reader Profile Page

rishithab48 opened the pull request:
- Added old profile page items to the newer one.
- Added and fixed the book stacks arrangement in the profile and book stacks page.
- Added a symbol button for removing books from book list instead of text button. (On profile page and book list page)
- Added a new logged out page.

**Fixed**
- Old profile page route.
- Removed home button on logged in version of homePage.
- Fixed some book list issues to show other users.

mutahharaj committed:
Merge branch 'develop' into mutahhara_jannat

rishithab48 committed:
Added old profile page items to new profile page.

rishithab48 committed:
Added bookList to profile page.

rishithab48 committed:
Added a button for removing books from database.

rishithab48 committed:
fixed booklist items alignment.

mutahharaj committed:
Merge branch 'develop' into mutahhara_jannat

rishithab48 committed:
Fixed booklists formatting. Made a new logged out page.

rishithab48 committed:
Merge branches 'rishitha_bura_profilepages' and 'develop' of https://github.com/WSU-4110/Novelology into rishitha_bura_profilepages

hello202320 commented:
I think we should of an profile page component --> author page or reader page to minimize code. Like OOP inheritance.

mutahharaj committed:
Settings Page Design

- worked on layout to match website design
- centered and aligned the features
- rearranged features
- resized and reshaped buttons, titles, forms,

rishithab48 commented:
So, make profile class the base class and author, reader profiles as their child classes?

mutahharaj committed:
Update Settings.js

mutahharaj committed:
Additional Changes to Settings

mutahharaj committed:
Settings Updated

- aligned role selection and account visibility buttons
- shortened desire username input/text box
-

rishithab48 committed:
Added ProfilePage and ReaderProfile classes.

rishithab48 committed:
Reader and Author Class added. Display published books component added.

racecn commented on src/App.js:
```diff
@@ -97,7 +97,8 @@ const App = () => {
           <Route path="/SearchResultsBook" element={<SearchResultsBook />} />
           <Route path="/ReaderQandA/:aid" element={<ReaderQandA />} />
-
+          <Route path='/loggedout' element = {<LoggedOutPage />}/>
```
Are these the permanent URLs we'd like to have? Or should we set the rendering choice within the home page component like before?

rishithab48 replied:
No, I am just using it to have a separate page when a user logged out and redirect them to logged out home page.

rishithab48 replied:
logged out page is to make sure that users logged out properly and to get back to the logged out navigation bar.

racecn commented on src/components/BookStacks/DisplayBookLists.js:
```diff
@@ -182,21 +195,41 @@ export const DisplayBookListItems = ({ bookData, userAuth }) => {
                   {book ? (
                     <div key={index}>
                       <img
+                        style={{ width: "96px", height: "144px" }}
```
Have to be careful with pixel specific styling, maybe like 10em by 15 em?

rishithab48 replied:
Changed it to rem.

rishithab48 committed:
Merge branch 'mutahhara_jannat' of https://github.com/WSU-4110/Novelology into rishitha_bura_profilepages

racecn commented on src/components/BookStacks/DisplayBookLists.js:
```diff
@@ -209,40 +242,222 @@ export const DisplayBookListItems = ({ bookData, userAuth }) => {
+              <div className="mt-4 text-center">
+                <button
+                  onClick={toggleShowAll}
+                  className="bg-maroon hover:bg-opacity-50 text-white font-bold py-2 px-4 rounded"
```
can we make a color in the tailwind config that represents this 50% opacity maroon color, I think we are using it in a few other places like the UploadPFP component

rishithab48 replied:
I found the color and added it to tailwind.config.js. UploadPFP has a different shade of it. I will find the color and change it.

racecn commented on src/pages/Settings.js:
```diff
@@ -261,48 +261,35 @@ export default function Settings() {
     
                                 <button
-                                    className="bg-red-500 hover:bg-red-600 text-white font-bold py-2 px-4 rounded mt-4"
+                                    className="bg-maroon hover:bg-opacity-70 text-white font-bold py-2 px-4 rounded-full mt-4"
```
Can we make a tailwind entry for this color, like light-maroon?

rishithab48 replied:
I added the color to tailwing.config.js and changed it here.

racecn commented on src/functions/Profile/ProfilePage.js:
```diff
@@ -0,0 +1,136 @@
+            this.followingCount = this.userData.following?.length || 0;
+            this.isFollowing = this.userData.following?.includes(this.uid);
+            //    userData.followers?.includes(currentUser.uid)
+            console.log("user data fetched successfully");
```
These logs seem like they are performing the same action as the try catch function that you have wrapping this get func

rishithab48 replied:
I removed the else. Try-catch is used to catch any exceptions in this case.

racecn commented on src/functions/AddBook.js:
```diff
@@ -8,12 +8,14 @@ export const AddBookToFirestore = async (BookData,navigate) => {
   if (BookData){
   const isbn = GetISBN13(BookData);
   const book_volumeInfo = BookData.volumeInfo;
+  const book_saleInfo = BookData.saleInfo;
```
Is sale info the price? or is this the location where you can buy it?

rishithab48 replied:
saleInfo has the price information and purchase links from google books. Mutahhara is working on it.

racecn commented on src/components/user/SelectGenres.js:
```diff
@@ -79,7 +79,7 @@ function SelectGenres() {
   return (
     <>
-      <div className="flex flex-col rounded-lg shadow-2xl items-center px-4 pt-7 pb-16 mt-6 max-w-full text-xl text-center text-black shadow-sm bg-maroon bg-opacity-50 w-[908px] max-md:px-5">
+      <div className="flex flex-col rounded-lg shadow-2xl items-center px-4 pt-7 pb-16 mt-6 max-w-full w-[908px] text-xl text-center text-black shadow-sm bg-maroon bg-opacity-50 max-md:px-5">
```
can we make this a non-pixel based width?

rishithab48 replied:
I changed it to rem

rishithab48 committed:
pull request fixes.
